pkg/controller/machine/machine_controller.go

```diff
@@ -180,7 +180,14 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 			return reconcile.Result{}, err
 		}
 
-		if m.Status.NodeRef != nil {
+		okToDeleteNode, err := r.isDeleteNodeAllowed(context.Background(), m)
+		if err != nil {
+			klog.Errorf("IsDeleteNodeAllowed check failed for machine %q: %v", name, err)
+			return reconcile.Result{}, err
+		}
+		if !okToDeleteNode {
+			klog.Infof("Deleting node is not allowed for machine %q", name)
```

Comment:
Hi @vincepri / @ncdc / @detiber,

I'm wondering if I shouldn't omit the log here and move it to `isDeleteNodeAllowed` in order to log the reason the node was not deleted. I followed the pattern from above, but maybe here it's worth adding this detail?

Reply:
+1 for moving

Reply:
Hi @ncdc,

I just refactored things a bit to improve the logging per this feedback. Please take a look.